src/Pelagos/Entity/DatasetSubmission.php

```diff
@@ -1194,6 +1194,7 @@ public function reviewEvent(Person $reviewer, $eventName)
                 //Setting the status to in-review.
                 $this->status = self::STATUS_COMPLETE;
                 $this->setDatasetStatus(Dataset::DATASET_STATUS_ACCEPTED);
+                $this->getDataset()->setAcceptedDate(new \DateTime('now', new \DateTimeZone('UTC')));
```

Comment:
I feel these three calls should move Dataset Entity, and just have one function name acceptDataset that can check if a dataset is in state to be acceptable, and then set the date, and status of itself. Dataset status and setAcceptedDate should be private functions. However, then there should also be function called rejectDataset and such. Maybe an improvement?

Reply:
Yeah, it think you are right. Previously we had the status tied up to the DatasetSubmission entity, now we've moved it to the dataset. Should we add it in the issues in github? to keep track of it.